mirdata/datasets/tiny_4mula.py

```diff
@@ -0,0 +1,319 @@
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets/tiny_4mula`
+
+    Attributes:
+        music_id (str): track id
+        music_name (str): music name
+        music_lang (str): music language
+        music_lyrics (str): lyrics
+        art_id (str): artist id
+        art_name (str): artist name
+        art_rank (int): artist rank
+        main_genre (str): artist main genre
+        related_genre (list): artist related genre
+        related_art (list): others artist related to artist
+        related_music (list): others music related to track
+        musicnn_tags (list): top-3 tags got by Musicnn
+        melspectrogram (numpy.ndarray): melspectrogram extracted by librosa
+
+
+    Cached Properties:
+        melspectrogram: melspectrogram in npy file
+        annotation: csv file with annotated metadata (all attributes, except melspectrogram)
+
+    """
+
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+
+        self.annotation_path = os.path.join(
+            self._data_home, self._track_paths["annotation"][0]
+        )
+        self.melspectrogram_path = os.path.join(
+            self._data_home, self._track_paths["melspectrogram"][0]
+        )
+
+    @property
+    def music_id(self):
+        return self._track_metadata.get("music_id")[0]
+
+    @property
+    def music_name(self):
+        return self._track_metadata.get("music_name")[0]
+
+    @property
+    def music_lang(self):
+        return self._track_metadata.get("music_lang")[0]
+
+    @property
+    def music_lyrics(self):
+        return self._track_metadata.get("music_lyrics")[0]
+
+    @property
+    def art_id(self):
+        return self._track_metadata.get("art_id")[0]
+
+    @property
+    def art_name(self):
+        return self._track_metadata.get("art_name")[0]
+
+    @property
+    def art_rank(self):
+        return self._track_metadata.get("art_rank")[0]
+
+    @property
+    def main_genre(self):
+        return self._track_metadata.get("main_genre")[0]
+
+    @property
+    def related_art(self):
+        return eval(self._track_metadata.get("related_art")[0])
+
+    @property
+    def related_genre(self):
+        return eval(self._track_metadata.get("related_genre")[0])
+
+    @property
+    def related_music(self):
+        return eval(self._track_metadata.get("related_music")[0])
+
+    @property
+    def musicnn_tags(self):
+        return eval(self._track_metadata.get("musicnn_tags")[0])
+
+    @property
+    def _track_metadata(self):
+        metadata = read_csv(self.annotation_path, sep="\t")
+        if not metadata.empty:
+            return metadata
+        return None
```

Comment:
This is already [implemented in core](https://github.com/mir-dataset-loaders/mirdata/blob/master/mirdata/core.py#L409-L416) and inherited by your track class here, so no need to implement again. Note that `_track_metadata` gets the metadata of each track from the `_metadata` property of the dataset, and that is the one you should customize to make sure that `_metadata` returns a dictionary with the `track_ids` as `keys`. Let me know if it's not clear!